apps/rush-lib/src/logic/base/BaseInstallManager.ts

```diff
@@ -263,12 +264,39 @@ export abstract class BaseInstallManager {
     shrinkwrapFile: BaseShrinkwrapFile | undefined
   ): Promise<{ shrinkwrapIsUpToDate: boolean; shrinkwrapWarnings: string[] }>;
 
-  protected abstract canSkipInstall(lastInstallDate: Date): boolean;
-
   protected abstract installAsync(cleanInstall: boolean): Promise<void>;
 
   protected abstract postInstallAsync(): Promise<void>;
 
+  protected canSkipInstall(lastModifiedDate: Date): boolean {
+    // Based on timestamps, can we skip this install entirely?
+    const potentiallyChangedFiles: string[] = [];
+
+    // Consider the timestamp on the node_modules folder; if someone tampered with it
+    // or deleted it entirely, then we can't skip this install
+    potentiallyChangedFiles.push(
+      path.join(this.rushConfiguration.commonTempFolder, RushConstants.nodeModulesFolderName)
+    );
+
+    // Additionally, if they pulled an updated npm-shrinkwrap.json file from Git,
+    // then we can't skip this install
+    potentiallyChangedFiles.push(this.rushConfiguration.getCommittedShrinkwrapFilename(this.options.variant));
+
+    // Add common-versions.json file to the potentially changed files list.
+    potentiallyChangedFiles.push(this.rushConfiguration.getCommonVersionsFilePath(this.options.variant));
+
+    if (this.rushConfiguration.packageManager === 'pnpm') {
+      // If the repo is using pnpmfile.js, consider that also
+      const pnpmFileFilename: string = this.rushConfiguration.getPnpmfilePath(this.options.variant);
+
+      if (FileSystem.exists(pnpmFileFilename)) {
```

Comment:
What if the pnpmfile was deleted? Shouldn't that case be considered a change?

Reply:
It would be considered a change. However, there's no way for us to know how to handle that without calling back to `git` unless we kept a list of files that "potentially changed" from the last time and diffed against it. Which is a good solution, but out of scope for this PR (all this change did was move the existing implementation)